frontend/src/router/index.ts

```diff
@@ -69,11 +75,19 @@ const routes = [
         component: ManagePermissionsView,
         beforeEnter: async (to: any) => {
             // Requires fetching applications the user administers.
-            await fetchApplications();
-            const userRoleAssignments = await fetchUserRoleAssignments(
-                selectedApplication.value?.application_id
-            );
-            Object.assign(to.meta, { userRoleAssignments: userRoleAssignments });
+            try {
```

Comment:
Hi @MCatherine1994 ,
We want to avoid using `try-cath` in router transition, it is not scalable for `try-catch` in every route (just same as you already saw in backend `router` for audit logging), and isn't the intention from Vue Router lifecycle hook.
In this case, if we try and catch, then when error happens, it will still land on `next` route. That's exactly what router guards trying to prevent: **only go to new route component when everything is fine or user is permitted, otherwise, not at all and stay in previous route**.

I see you also refactor ToastMessage to make message working when error is happening in routing. Thanks for the good refactoring! Now I kind of understand what's going why toast message wasn't showing previously. I will comment on the ToastState.ts since you have a good find actually!!

However, we can talk about this later. I will take a deeper look on this.